### PR TITLE
Add hurricane swap strategies

### PR DESCRIPTION
**Pull request by simednba**
Add HurricaneSwap strategies for LP & XHCT.
These strategies use the Hurricane Router instead of swapping directly through the pair because the swap function of a pair is onlyOnwer.

### CODE REVIEW

File: contracts/strategies/HurricaneStratForLP.sol [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+    updateDevFee(_devFeeBips);
+    updateReinvestReward(_reinvestRewardBips);
+    updateDepositsEnabled(true);
+    transferOwnership(_timelock);
+    
+
+    emit Reinvest(0, 0);
+  }
+
+  function setRoutes(address[] memory _route0, address[] memory _route1) external onlyOwner {
```
Comment (ASlowCheetah) [inline contracts/strategies/HurricaneStratForLP.sol:61]: this needs to be onlyDev, owner and dev could be different persons with different timelocks

File: contracts/strategies/HurricaneStratForLP.sol [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+    require(amount > 0, "HurricaneStrategyForLP::amount=0");
+    stakingContract.deposit(PID, amount);
+  }
+
+  function checkReward() public override view returns (uint) {
+    (uint256 pendingAmounts,) = stakingContract.pending(PID, address(this));
+    return pendingAmounts;
+  }
+
+    function getFinalAmountOut(address factory, uint amountIn, address[] memory path) internal view returns (uint amount) {
```
Comment (ASlowCheetah) [inline contracts/strategies/HurricaneStratForLP.sol:190]: some tab issues here, formatting, could you run prettier on the files so we get it consistent?

```
npx prettier --write "contracts/strategies/HurricateStratForLP.sol"
```

If you could, please also format the other files in this PR

**Review by ASlowCheetah — COMMENTED** [review]
Changes looks good, did you run any test on these? If so would you mind sharing? Don't need to be part of the PR a git gist is enough

**Comment by simednba** [comment]
tests written ( on avax forked chain, with python/brownie) : https://gist.github.com/simednba/b77df5d505828d8179a1e3c46472e271

**Pull request merged**